joblib/numpy_pickle.py

```diff
@@ -6,32 +6,27 @@
 # Copyright (c) 2009 Gael Varoquaux
 # License: BSD Style, 3 clauses.
 
+import os
 import pickle
-import traceback
 import sys
-import os
-import zlib
+import traceback
 import warnings
-import struct
-import codecs
+import zlib
+from io import BytesIO
 
 from ._compat import _basestring
-
-from io import BytesIO
+from .externals import dill
 
 PY3 = sys.version_info[0] >= 3
+Pickler = dill.Pickler
+Unpickler = dill.Unpickler
```

Comment:
Maybe import Pickler and Unpickler using a single line import and remove line18 above:
`from .externals.dill import Pickler, Unpickler`